[CI] SearchStatesIT testCanMatch failing

```
Caused by:	
        java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-9 [ACTIVE]	
            at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)	
            at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:98)	
            at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:40)	
            at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)	
            at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)	
            at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502)	
            at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)	
            at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)	
            at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)	
            at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)	
            at java.base/java.lang.Thread.run(Thread.java:1583)
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/5g73xtvoexkzs/tests/:qa:ccs-rolling-upgrade-remote-cluster:v8.10.4%23oldClusterTest/org.elasticsearch.upgrades.SearchStatesIT/testCanMatch


**Reproduction line:**
```
./gradlew ':qa:ccs-rolling-upgrade-remote-cluster:v8.10.4#oldClusterTest' -Dtests.class="org.elasticsearch.upgrades.SearchStatesIT" -Dtests.method="testCanMatch" -Dtests.seed=B61BF2F991120A4D -Dtests.bwc=true -Dtests.locale=ja -Dtests.timezone=Europe/Gibraltar -Druntime.java=21
```

**Applicable branches:**
~~8.12~~ main

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.upgrades.SearchStatesIT#testCanMatch`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testCanMatch'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.upgrades.SearchStatesIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-9 [ACTIVE]

  at __randomizedtesting.SeedInfo.seed([B61BF2F991120A4D:A03FC7B7C0A1A664]:0)
  at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:919)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
  at org.elasticsearch.test.rest.ESRestTestCase.createIndex(ESRestTestCase.java:1687)
  at org.elasticsearch.test.rest.ESRestTestCase.createIndex(ESRestTestCase.java:1648)
  at org.elasticsearch.upgrades.SearchStatesIT.testCanMatch(SearchStatesIT.java:214)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-9 [ACTIVE]

    at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
    at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:98)
    at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:40)
    at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
    at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
    at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502)
    at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
    at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
    at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
    at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
    at java.lang.Thread.run(Thread.java:1583)

```

The underlying error is an unknown field "lifecyle" for index template. The same failure happens for BWC with [8.9.2](https://gradle-enterprise.elastic.co/s/6bzepyrsypkvs), [8.10.0](https://gradle-enterprise.elastic.co/s/v6ctf4sd5pk6q) and other patch releases.
 
```
» [2023-12-12T21:13:40,629][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [v8.10.4-local-1] fatal error in thread [elasticsearch[v8.10.4-local-1][cluster_coordination][T#1]], exiting java.lang.AssertionError: org.elasticsearch.gateway.CorruptStateException: org.elasticsearch.xcontent.XContentParseException: [-1:23997] [index_template] failed to parse field [index_template]
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.assertOnCommit(PersistedClusterStateService.java:1244)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.commit(PersistedClusterStateService.java:1234)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.writeIncrementalStateAndCommit(PersistedClusterStateService.java:976)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.GatewayMetaState$LucenePersistedState.writeClusterStateToDisk(GatewayMetaState.java:579)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.GatewayMetaState$LucenePersistedState.setLastAcceptedState(GatewayMetaState.java:562)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.CoordinationState.handlePublishRequest(CoordinationState.java:392)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.Coordinator.handlePublishRequest(Coordinator.java:469)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.PublicationTransportHandler.acceptState(PublicationTransportHandler.java:213)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.PublicationTransportHandler.handleIncomingPublishRequest(PublicationTransportHandler.java:200)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.PublicationTransportHandler.lambda$new$0(PublicationTransportHandler.java:112)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:74)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:294)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
»  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
»  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
»  	at java.base/java.lang.Thread.run(Thread.java:1583)
»  Caused by: org.elasticsearch.gateway.CorruptStateException: org.elasticsearch.xcontent.XContentParseException: [-1:23997] [index_template] failed to parse field [index_template]
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:674)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.lambda$loadOnDiskState$6(PersistedClusterStateService.java:572)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.consumeFromType(PersistedClusterStateService.java:716)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.loadOnDiskState(PersistedClusterStateService.java:571)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.assertOnCommit(PersistedClusterStateService.java:1242)
»  	... 16 more
»  Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:23997] [index_template] failed to parse field [index_template]
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.ComposableIndexTemplateMetadata.fromXContent(ComposableIndexTemplateMetadata.java:67)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:54)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:147)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:414)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.Metadata$Builder.fromXContent(Metadata.java:2660)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:672)
»  	... 20 more
»  Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:23997] [index_template] failed to parse field [template]
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.ComposableIndexTemplate.parse(ComposableIndexTemplate.java:102)
»  	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.ComposableIndexTemplateMetadata.lambda$static$1(ComposableIndexTemplateMetadata.java:50)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
»  	... 29 more
»  Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:23995] [template] unknown field [lifecycle]
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.lambda$errorOnUnknown$2(ObjectParser.java:105)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:299)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431)
»  	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
»  	... 36 more
```

--- a/qa/ccs-rolling-upgrade-remote-cluster/build.gradle
+++ b/qa/ccs-rolling-upgrade-remote-cluster/build.gradle
@@ -6,6 +6,7 @@
  * Side Public License, v 1.
  */
 
+import org.elasticsearch.gradle.Version
 import org.elasticsearch.gradle.internal.info.BuildParams
 import org.elasticsearch.gradle.testclusters.StandaloneRestIntegTestTask
 
@@ -32,6 +33,7 @@ BuildParams.bwcVersions.withWireCompatible { bwcVersion, baseName ->
     versions = [bwcVersion.toString(), project.version]
     setting 'cluster.remote.node.attr', 'gateway'
     setting 'xpack.security.enabled', 'false'
+    requiresFeature 'es.failure_store_feature_flag_enabled', new Version(8, 12, 0)
   }
   def remoteCluster = testClusters.register("${baseName}-remote") {
     numberOfNodes = 3
@@ -39,6 +41,7 @@ BuildParams.bwcVersions.withWireCompatible { bwcVersion, baseName ->
     firstNode.setting 'node.attr.gateway', 'true'
     lastNode.setting 'node.attr.gateway', 'true'
     setting 'xpack.security.enabled', 'false'
+    requiresFeature 'es.failure_store_feature_flag_enabled', new Version(8, 12, 0)
   }
 
 

